flowauth/backend/flowauth/token_management.py

```diff
@@ -153,28 +142,57 @@ def add_token(server):
     latest_lifetime = current_user.latest_token_expiry(server)
     if expiry > latest_lifetime:
         raise InvalidUsage("Token lifetime too long", payload={"bad_field": "expiry"})
-    allowed_claims = current_user.allowed_claims(server)
 
-    current_app.logger.debug("New token request", allowed_claims=allowed_claims)
-    for claim in json["claims"]:
-        if claim not in allowed_claims:
-            raise Unauthorized(f"You do not have access to {claim} on {server.name}")
```

Comment:
Need an equivalent of this in the new roles-based `add_token()`. I.e. `allowed_roles = current_user.allowed_roles(server)`, and check that all requested roles are in `allowed_roles`. Specifically, this needs to check:
1. Current user has access to `role`, and
2. `role` corresponds to the correct server (i.e. ensure we're not including scopes from a role on server B in a token valid for server A)

Reply:
This is in-place with a test, `test_reject_claims_when_not_allowed`

Reply:
Would be good to have a test for point 2 as well (i.e. reject if current user does have access to `role`, but `role` corresponds to the wrong server) - that's the kind of thing I can imagine we could easily break in future changes if we don't have a test for it.

Reply:
Case 2 now tested in the same test